modules/codegen-plugin/src/sbt-test/codegen-plugin/smithy-build/smithy-build.json

```diff
@@ -0,0 +1,12 @@
+{
+  "version": "1.0",
+  "plugins": {
+    "openapi": {
+      "service": "smithy4s.example#ObjectService",
```

Comment:
our of curiosity, does this imply that the openapi-translation of only one service can be configured this way ?

Reply:
The `alloy` conversion logic completely ignores whatever service is set here and will apply [the same OpenAPI config to all services](https://github.com/disneystreaming/alloy/blob/main/modules/openapi/src/alloy/openapi/package.scala#L79-L81). If we wanted to do anything involving service-based filtering, that would have to change first. For now, it's probably worth mentioning this and other Alloy OpenAPI overrides in the docs - I can add a bit about that.

That said, good point that if alloy _did_ check the service name, the current implementation would only allow for one service's OpenAPI to be configured. Checking the Smithy docs, it looks like I overlooked [support for an optional artifact ID after the plugin name](https://smithy.io/2.0/guides/smithy-build-json.html#plugin-id). If I fix that, you'd be able to include separate plugins `openapi::foo` vs `openapi::bar`, but you'd need a change on the alloy side to be able to actually pass multiple of them and have them be respected.

Reply:
Alright, that's somewhat disappointing. If it comes to that, I suppose we could codify our own plugin semantics (which could map a unique configuration to services). 

Regardless, this doesn't take away from the fact that this PR is a decent step forward 👍